main.py

```diff
@@ -200,10 +210,12 @@ def get_summary(txt, lm):
   inventory
 
 seldocs = select_docs(dt, ch, lg, lm, ck, ct)
+summaries = gather_summaries(dt, ch, lg, lm, ck, ct, seldocs)
+st.success('Summaries loaded!')
 
-for d in seldocs:
+for i, d in enumerate(seldocs):
   try:
-    smr = get_summary(d.page_content, lm)
+    smr = json.loads(summaries[i])
```

Comment:
Is the order of input iterable and results maintained in the thread pool function used? If so, then it is okay, otherwise metadata and summary might be mixed up. I am not sure about it, so please look at docs and confirm the behavior.

Reply:
Yes, ordering is preserved ([source](https://superfastpython.com/threadpool-starmap/))